state/state_test.go

```diff
@@ -130,80 +131,53 @@ func TestValidatorSimpleSaveLoad(t *testing.T) {
 	assert.IsType(ErrNoValSetForHeight{}, err, "expected err at unknown height")
 }
 
-// TestValidatorChangesSaveLoad tests saving and loading a validator set with changes.
+// TestValidatorChangesSaveLoad tests saving and loading a validator set with
+// changes.
 func TestValidatorChangesSaveLoad(t *testing.T) {
+	const valSetSize = 7
 	tearDown, _, state := setupTestCase(t)
+	state.Validators = genValSet(valSetSize)
+	state.Save()
 	defer tearDown(t)
-	// nolint: vetshadow
-	assert := assert.New(t)
 
-	// change vals at these heights
-	changeHeights := []int64{1, 2, 4, 5, 10, 15, 16, 17, 20}
```

Comment:
why remove this test ?

Reply:
because it was written with an assumption that there is only one validator. I can try and rewrite it of course

Reply:
right. I think it's ok since we're specifically testing the "save/load at height" behaviour, and having just one validator was an easy way to write the test, and we have other tests of multiple validators. So we should defntly keep it. If you have a nice way to write it with multiple validators, that'd of course be welcome, though now we have to be mindful that it can only change by < 1/3 :)